File: bumperbot_localization/bumperbot_localization/imu_republisher.py
#!/usr/bin/env python3

import rclpy
from rclpy.node import Node
from sensor_msgs.msg import Imu
import time


imu_pub = None

def imu_callback(msg):
    global imu_pub
    msg.header.frame_id = "base_footprint_ekf"
    imu_pub.publish(msg)
  
def main():
    global imu_pub
    rclpy.init()
    node = Node("imu_republisher")
    time.sleep(1)
    imu_pub = node.create_publisher(Imu, "imu_ekf", 10)
    imu_sub = node.create_subscription(Imu, "imu/out", imu_callback, 10)
    rclpy.spin(node)
    node.destroy_node()
    rclpy.shutdown()


if __name__ == "__main__":
    main()

File: bumperbot_localization/src/imu_republisher.cpp
#include<rclcpp/rclcpp.hpp>
#include<sensor_msgs/msg/imu.hpp>

using namespace std::chrono_literals;

rclcpp::Publisher<sensor_msgs::msg::Imu>::SharedPtr imu_pub;

void imu_callback(const sensor_msgs::msg::Imu &imu_msg)
{
    sensor_msgs::msg::Imu new_imu_msg;
    new_imu_msg = imu_msg ;
    new_imu_msg.header.frame_id = "base_footprint" ;
    imu_pub->publish(new_imu_msg);
}

int main(int argc, char* argv[])
{
    
    rclcpp::init(argc, argv);
    auto node = rclcpp::Node::make_shared("imu_republisher");
    rclcpp::sleep_for(1s);
    imu_pub = node->create_publisher<sensor_msgs::msg::Imu>("imu_ekf", 10);
    auto imu_sub = node->create_subscription<sensor_msgs::msg::Imu>("imu/out", 10, imu_callback);
    rclcpp::spin(node);
    rclcpp::shutdown();
    
    return 0;
}